ext/standard/tests/strings/str_pad_variation1.phpt

```diff
@@ -0,0 +1,36 @@
+--TEST--
+Test str_pad() function : usage variations - large values for '$pad_length' argument
+--FILE--
+<?php
+/* Prototype  : string str_pad  ( string $input  , int $pad_length  [, string $pad_string  [, int $pad_type  ]] )
+ * Description: Pad a string to a certain length with another string
+ * Source code: ext/standard/string.c
+*/
+
+/* Test str_pad() function: with unexpected inputs for '$pad_length'
+ *  and expected type for '$input'
+*/
+
+echo "*** Testing str_pad() function: with large value for for 'pad_length' argument ***\n";
+
+//defining '$input' argument
+$input = "Test string";
+
+$extra_large_pad_length = PHP_INT_MAX*5;
+try {
+    var_dump( str_pad($input, $extra_large_pad_length) );
+} catch (\TypeError $e) {
+    echo $e->getMessage() . "\n";
+}
+
+$php_int_max_pad_length = PHP_INT_MAX;
+var_dump( str_pad($input, $php_int_max_pad_length) );
+
+
+?>
+--EXPECTF--
+*** Testing str_pad() function: with large value for for 'pad_length' argument ***
+str_pad() expects parameter 2 to be int, float given
+
+Fatal error: Allowed memory size of 134217728 bytes exhausted%s(tried to allocate %d bytes) in %s on line %d
```

Comment:
This should probably use `%d` for the bytes as a custom `php.ini` may be supplied to the `phpt`

Reply:
Right makes sense, I just copy pasted this from the variation5 test which defines a custom memory_limit.
```
--INI--
memory_limit=128M
```
Should I rather do that or replace it with ``%d`` ?

Reply:
Both should be just fine so no hard preference here from my side :)

Reply:
Replaced with ``%d`` :)